packages/ketcher-core/src/application/render/renderStruct.ts

```diff
@@ -24,6 +25,25 @@ export class RenderStruct {
     return struct;
   }
 
+  static removeSmallAttachemntPointLabelsInModal(render: Render, options) {
+    if (!options.labelInMonomerConnectionsModal) {
+      return;
+    }
+
+    const attachmentPointRegExp = /^R[1-8]$/;
+
+    render.ctab.atoms.forEach((atom: ReAtom) => {
+      if (!atom.label) {
+        return;
+      }
+      const isSmall = atom.label.path.node.getBoundingClientRect().width < 8;
```

Comment:
I would suggest to move it under "if" statement to get boundingClientRect only for attachmentPointAtoms because it is very unefficient method from the performance point of view.